keras/src/layers/layer.py

```diff
@@ -1206,6 +1231,37 @@ def _check_quantize_args(self, mode, compute_dtype):
                 "'mixed_float16' before calling `quantize()`."
             )
 
+    def quantized_call(self, *args, **kwargs):
+        if self.quantization_mode == "int8":
+            return self._int8_call(*args, **kwargs)
+        elif self.quantization_mode == "float8":
+            return self._float8_call(*args, **kwargs)
+        else:
+            raise self._quantization_mode_error(self.quantization_mode)
+
+    def _int8_call(self, *args, **kwargs):
+        raise self._not_implemented_error(self._int8_call)
+
+    def _float8_call(self, *args, **kwargs):
+        raise self._not_implemented_error(self._float8_call)
+
+    def _not_implemented_error(self, method):
+        if callable(method):
+            method_name = method.__name__
+        else:
+            method_name = str(method)
+        return NotImplementedError(
+            f"Layer {self.__class__.__name__} does not have a `{method_name}` "
+            "method implemented."
```

Comment:
Consider using:

```python
    def _raise_not_implemented_error(self, attr):
        if callable(attr):
            attr_name = attr.__name__
            attr_type = "method"
        else:
            attr_name = str(attr)
            attr_type = "attribute"
        return NotImplementedError(
            f"Layer {self.__class__.__name__} does not have a `{attr_name}` "
            f"{attr_type} implemented."
```

Reply:
I have extended it a bit and used it for almost all `NotImplementedError` in `Layer`:
```python
    def _not_implemented_error(self, attr, msg=None):
        if callable(attr):
            attr_name = attr.__name__
            attr_type = "method"
        else:
            attr_name = str(attr)
            attr_type = "attribute"
        msg = " " + msg if msg is not None else ""
        return NotImplementedError(
            f"Layer {self.__class__.__name__} does not have a `{attr_name}` "
            f"{attr_type} implemented.{msg}"
        )
```